### ISSUE TITLE
Manually downloading a LLM

### ISSUE THREAD

**Issue by ekuznetsov139** [pieces-app/support#527]
### Software

Desktop Application

### Operating System / Platform

Linux

### Your Pieces OS Version

Latest

### Early Access Program

- [ ] Yes, this is related to an Early Access Program feature.

### Kindly describe the bug and include as much detail as possible on what you were doing so we can reproduce the bug.

I am attempting to install Pieces with a local LLM. Unfortunately, I have an unstable network connection (which is why I want a local model in the first place.) I am trying to get Mistral-7B.

I found that Pieces wants to download the whole model (4 GB) all at once. If it fails for whatever reason, it restarts from the beginning.  This is annoying and it makes downloading virtually impossible, because my network connection does not stay long enough to pull 4GB in one go.

I have figured out where the file is being downloaded to and from, and managed to download it manually with wget. Now the problem is that Pieces refuses to recognize it. Even if I put it in the right place, the Pieces app stil shows the Download button in the UI, and clicking Download truncates the file to zero and starts the download from the beginning.

I tried to edit models.db and change "downloaded" to true in the relevant line, but it has no effect.

Is there anything else I could do?

**Comment by hra42** [comment]
I don't think you could import a model you downloaded manually. The problem is that pieces doesn't just download the raw model, but rather how the integration works.

@mark-at-pieces might be able to talk about this further.

**Comment by brian-at-pieces** [comment]
Hey @ekuznetsov139 sorry you're having this issue. The reason we delete the model on a failed download is becuase we're unable to determine whether the partial download was corrupted or not, so we had to fully delete to ensure success. 

Nice try on updating the db property, but that won't be enough. Try this:
1. Shutdown Pieces
2. Add an empty "download_success.txt" to the mistral directory in com.pieces.os/llm_chat_models Also check that your directory structure looks like the attached screenshot
3. Restart Pieces

We actually have a new release coming the next day or two that completely overhauls our on-device LLM infrastructure - including the ability to restart a failed download from where it left off. If you can hang on until then your problem will be resolved.

<img width="979" alt="Image" src="https://github.com/user-attachments/assets/60705f20-3a85-4405-8068-f1af12a09e5c" />

**Comment by brian-at-pieces** [comment]
@ekuznetsov139 really sorry for all the trouble here. I've opened a [new issue](https://github.com/pieces-app/support/issues/537) for the failure to install and I'm currently investigating. In the meantime, if you download Ollama yourself Pieces should be able to utilize it: https://ollama.com/download/linux

I'll get back to you ASAP. Thanks for the patience

**Comment by BeZazz** [comment]
Any updates, other then the will be fixed soon type replies? It has been months.
